File: MyList.Server/Models/ListItem.cs
using System.ComponentModel.DataAnnotations;

namespace MyList.Server.Models
{
    public class ListItem
    {
        [Key]
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public required UserList List { get; set; }
    }
}

File: mylist.search-engine/src/main/kotlin/xpa/shadow/mylist/data/ListItem.kt
package xpa.shadow.mylist.data

import jakarta.persistence.*
import java.io.Serializable
import java.time.Clock
import java.time.OffsetDateTime

@Entity(name = "Items")
class ListItem : Serializable {

    @Id
    var id: Int = 0
    var name: String? = null
    var description: String? = null
    var createdAt: OffsetDateTime = OffsetDateTime.now(Clock.systemUTC())
    var updatedAt: OffsetDateTime? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ListId")
    var list: UserList? = null
}
